specification/communication/data-plane/Identity/stable/2025-06-30/CommunicationIdentity.json

```diff
@@ -0,0 +1,695 @@
+      }
+    },
+    "CommunicationIdentityCreateRequest": {
+      "type": "object",
+      "properties": {
+        "createTokenWithScopes": {
+          "description": "Also create access token for the created identity.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/CommunicationIdentityTokenScope"
+          }
+        },
+        "expiresInMinutes": {
+          "format": "int32",
+          "description": "Optional custom validity period of the token within [60,1440] minutes range. If not provided, the default value of 1440 minutes (24 hours) will be used.",
+          "type": "integer",
+          "minimum": 60,
+          "maximum": 1440,
+          "default": 1440
+        }
+      }
+    },
+    "CommunicationIdentityTokenScope": {
+      "description": "List of scopes for an access token.",
+      "enum": [
+        "chat",
+        "voip",
+        "chat.join",
+        "chat.join.limited",
+        "voip.join"
+      ],
+      "type": "string",
+      "x-ms-enum": {
+        "name": "CommunicationIdentityTokenScope",
+        "modelAsString": true,
+        "values": [
+          {
+            "value": "chat",
+            "description": "Use this for full access to Chat APIs."
+          },
+          {
+            "value": "voip",
+            "description": "Use this for full access to Calling APIs."
+          },
+          {
+            "value": "chat.join",
+            "description": "Access to Chat APIs but without the authorization to create, delete or update chat threads."
+          },
+          {
+            "value": "chat.join.limited",
+            "description": "A more limited version of chat.join that doesn't allow to add or remove participants. Use this scope when the token bearer is not fully trusted, for example in guest scenarios."
+          },
+          {
+            "value": "voip.join",
+            "description": "Access to Calling APIs but without the authorization to start new calls."
+          }
+        ]
+      }
+    },
+    "TeamsUserExchangeTokenRequest": {
+      "description": "A request to create or update a Teams Phone assignment.",
+      "required": [
+        "appId",
+        "token",
+        "userId"
+      ],
+      "type": "object",
+      "properties": {
+        "token": {
+          "description": "Entra ID access token of a Teams User to acquire a new Communication Identity access token.",
+          "type": "string"
+        },
+        "appId": {
+          "description": "Client ID of an Entra ID application to be verified against the appid claim in the Entra ID access token.",
+          "type": "string"
+        },
+        "userId": {
+          "description": "Object ID of an Entra ID user (Teams User) to be verified against the oid claim in the Entra ID access token.",
+          "type": "string"
+        }
+      }
+    },
+    "TeamsExtensionExchangeTokenRequest": {
+      "type": "object",
+      "properties": {}
+    },
```

Comment:
Is this schema exists here for future expansion? There is currently no property in it.

Reply:
Yes, exactly. We want to prevent breaking changes in the case of future optional parameters. We expect future expansion of the scope of Teams Phone Extensibility feature and it may lead to the necessity to add some parameters to this call.